src/lib.rs

```diff
@@ -204,40 +203,31 @@ unsafe fn get_stack_size_internal(_thread_attr: *const pthread_attr_t) -> usize
     return 0;
 }
 
+struct NumThreads {
+    n_tc: usize,
+    n_fc: usize,
+}
+
 #[cold]
-unsafe fn get_num_threads(
-    c: *mut Rav1dContext,
-    s: *const Rav1dSettings,
-    n_tc: *mut c_uint,
-    n_fc: *mut c_uint,
-) {
-    static fc_lut: [u8; 49] = [
-        1, 2, 2, 2, 3, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 5, 5, 5, 5, 5, 6, 6, 6, 6, 6,
-        6, 6, 6, 6, 6, 6, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7,
-    ];
-    *n_tc = (if (*s).n_threads != 0 {
-        (*s).n_threads
+fn get_num_threads(s: &Rav1dSettings) -> NumThreads {
+    let n_tc = if s.n_threads != 0 {
+        s.n_threads as usize
     } else {
-        iclip(rav1d_num_logical_processors(c), 1 as c_int, 256 as c_int)
-    }) as c_uint;
-    *n_fc = if (*s).max_frame_delay != 0 {
-        cmp::min((*s).max_frame_delay as c_uint, *n_tc)
+        rav1d_num_logical_processors().clamp(1, 256)
+    };
+    let n_fc = if s.max_frame_delay != 0 {
+        cmp::min(s.max_frame_delay as usize, n_tc)
     } else {
-        (if *n_tc < 50 as c_uint {
-            fc_lut[(*n_tc).wrapping_sub(1 as c_int as c_uint) as usize] as c_int
-        } else {
-            8 as c_int
-        }) as c_uint
```

Comment:
What happened to this bit of logic? It looks like you completely changed the logic here in the cleaned up version.

Reply:
Yeah, I simplified it in 93f4416.  That's what the comments said it was doing, so I did that using a sqrt instead of a lookup table which is IMO a lot more confusing.